src/Compilers/CSharp/Portable/Binder/WithLambdaParametersBinder.cs

```diff
@@ -28,7 +28,10 @@ public WithLambdaParametersBinder(LambdaSymbol lambdaSymbol, Binder enclosing)
                 RecordDefinitions(parameters);
                 foreach (var parameter in lambdaSymbol.Parameters)
                 {
-                    this.parameterMap.Add(parameter.Name, parameter);
+                    if (!(parameter is IDiscardSymbol))
```

Comment:
wait.... the parameter *itself* is a discard symbl and not an IParameterSymbol?  that seems rough...

Reply:
Well, as you noted below, it's both an `IParameterSymbol` or an `IDiscardSymbol`.

I'm not sure what would be some alternatives, maybe:
1. having it be an `IParameterSymbol` and having a flag that indicates that it's a discard
2. adding a new interface, `IDiscardParameterSymbol` (derived from `IParameterSymbol`)

I prefer the current design, but I'm open to feedback.

Reply:
I prefer '1'.  Parameters are simple enough to just want one IParameterSymbol to represent all of them for me.  A tiny knob like this feels much nicer as just a bool/flag on it IMO :)

Reply:
Agreed with Cyrus here. The only odd thing will be that `IParameterSymbol.IsDiscard` will be false when there's only 1 underscore, but we don't really have the option of changing that.

Reply:
If we decide that we do want to have an IParameterSymbol that is also an IDiscardSymbol in the future, we can always add the interface back without a breaking change.

Reply:
> 

Agreed with Cyrus here. The only odd thing will be that IParameterSymbol.IsDiscard will be false when there's only 1 underscore, 


Which actually makes sense to me :)    It's *not* a discard in that scenario as you *can* write  `_ => _`.  So i think this really accurately describes how the langauge sees things! :)

> but we don't really have the option of changing that.

Well, not without a breaking change in the future.  I could see us doing that.  But that's out of scope here.  Note that if we do change that, this API approach would still work.  it would just be that `_ => _` would become an error and the API would now return `true` for `.IsDiscard` here (which would be fine by me).

Reply:
Yes, we can implement `IDiscardSymbol` later on, but we'll be left with the `IsDiscard` API laying around.
Anyways, I think that approach is okay (don't implement `IDiscardSymbol`). Made the change.